generals/envs/gymnasium_generals.py

```diff
@@ -69,10 +69,10 @@ def reset(
             options = {}
 
         if "grid" in options:
-            grid = self.grid_factory.grid_from_string(options["grid"])
+            grid = Grid(options["grid"])
         else:
-            self.grid_factory.rng = self.np_random
-            grid = self.grid_factory.grid_from_generator()
+            self.grid_factory.set_seed(seed)
```

Comment:
This should be as before. When we subclass `gymnasium.Env`, new rng is created after every call to `reset()`, see [here](https://github.com/Farama-Foundation/Gymnasium/blob/e73245912087d47b538dcdb45fa9a9d185b805c5/gymnasium/core.py#L156). This generator should be then passed to all the components to ensure maximum reproducibility.

Reply:
Ahhhh! Gotcha. A bit tricky that the PettingZoo & Gymnasium reset API's differ in that way too eh.

Reply:
Yeah :/ Generally, I feel like pettingzoo is a bit underdeveloped, but better for multi agent RL. The idea behind also having gymnasium is that hopefully more agents will be published to this repository and can be then used as "environment NPCs" against which you can then train. I'll hopefully publish some vanilla supervised baseline in next 2 weeks or so.

Reply:
Mmmm gotcha